networkit/cpp/embedding/Node2Vec.cpp

```diff
@@ -31,9 +31,20 @@ namespace NetworKit {
 
 Node2Vec::Node2Vec(const Graph &G, double P, double Q, count L, count N, count D)
     : G(&G), P(P), Q(Q), L(L), N(N), D(D) {
+    if (G.isDirected()) {
+        throw std::runtime_error("Current implementation can only deal with undirected graphs");
+    }
     if (G.numberOfNodes() != G.upperNodeIdBound()) {
         throw std::runtime_error("The node ids of the graph must be continuous.");
     }
+    bool hasIsolatedNodes = false;
+    G.parallelForNodes([&](const node u) {
+        if (G.degree(u) == 0) {
+            hasIsolatedNodes = true;
+        }
+    });
+    if (hasIsolatedNodes)
+        throw std::runtime_error("Isolated nodes are not allowed.");
```

Comment:
It's probably cleaner to use:
```suggestion
    for (node u : G.nodeRange())
        if (G.degree(u) == 0)
            std::runtime_error("Isolated nodes are not allowed.");
```

Reply:
Exchanged `parallelForNodes` for `nodeRange`. However, testing for throws fails if the runtime_error is triggered inside a for-loop.